### ISSUE TITLE
Add support for tagging parameters

### ISSUE THREAD

**Issue by martincostello** [aws/aws-ssm-data-protection-provider-for-aspnet#45]
### Describe the feature

Add the ability to configure tag(s) to apply to parameters that are created in the parameter store.

### Use Case

To make it easier to track resources in the Parameter Store which are created by applications using the NuGet package.

### Proposed Solution

Extend `PersistOptions` with an `IDictionary<string, string>` (or similar) which is used to apply any user-specified tags to created parameters.

```diff
namespace Amazon.AspNetCore.DataProtection.SSM;

 public class PersistOptions
 {
     public string KMSKeyId { get; set; }
     public TierStorageMode TierStorageMode { get; set; } = TierStorageMode.StandardOnly;
+    public IDictionary<string, string> Tags { get; set; } = new Dictionary<string, string>();
 }
```

### Acknowledgements

- [X] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### AWS .NET SDK and/or Package version used

Amazon.AspNetCore.DataProtection.SSM 3.0.0.

### Targeted .NET Platform

.NET 7

### Operating System and version

Linux 5.4.172-90.336.amzn2.x86_64 #1 SMP Wed Jan 19 23:08:01 UTC 2022

**Comment by ashishdhingra** [comment]
@martincostello Thanks for opening feature request. Please confirm if you are referring to [Tagging Systems Manager parameters](https://docs.aws.amazon.com/systems-manager/latest/userguide/tagging-parameters.html#tagging-parameters-new) and adding support to the library.

**Comment by martincostello** [comment]
Yes, that's correct. I saw that the library created new parameters for the keys at runtime, but when viewing them in the AWS Console they had no tags.

**Comment by saviour123** [comment]
Can the tagging feature be made optional like the .ApplicaitonName()

**Comment by normj** [comment]
@martincostello I think this sounds like a great feature idea. If you are interested in doing a PR for it that is more likely to make it get done and released sooner. Otherwise we can add it to our queue. If you do make a PR feel free to tag me.

@saviour123 I'm not sure what you mean by optional. If the `Tags` collection is empty then no tags would be applied to the parameter.

**Comment by martincostello** [comment]
Thanks Norm - I’ll see if I can make a start on something next week.

**Comment by martincostello** [comment]
@normj Opened #48 to implement this.

**Comment by normj** [comment]
Feature is released as part of version 3.1.0
